From 6882302cd00eb7749c01e25e2478a832f5f8552f Mon Sep 17 00:00:00 2001
From: Wan-Teh Chang <wtc@google.com>
Date: Tue, 16 Aug 2022 16:38:05 +0000
Subject: [PATCH] Roll src/third_party/libavif/src/ dd2d67c5f..d22d4de94 (35
 commits)

https://chromium.googlesource.com/external/github.com/AOMediaCodec/libavif.git/+log/dd2d67c5f976..d22d4de94120

$ git log dd2d67c5f..d22d4de94 --date=short --no-merges --format='%ad %ae %s'
2022-08-11 vigneshv Refactor libyuv yuv->rgb conversion with function pointers
2022-08-10 vigneshv Use per-row conditional in avifImageIdentity8ToRGB8Color
2022-08-11 wtc Turn on caching of Build SVT-AV1 and Build libjpeg
2022-08-11 vigneshv Update libyuv.cmd to 9b17af9b
2022-08-10 wtc ext/svt.sh: Do not build the SVT-AV1 decoder
2022-08-10 wtc Ext: Update to SVT-AV1 v1.2.0
2022-08-09 vigneshv Support YUV422->RGB565 conversion with libyuv
2022-08-10 yguyon Run ext/libjpeg.cmd in GitHub CI
2022-08-04 vigneshv Add support RGB565 format
2022-08-09 wtc avifPNGWrite: Output RGB if input has no alpha
(...)
2022-08-02 wtc Fix MSVC warnings
2022-08-01 yguyon Make avifImageAllocatePlanes() return avifResult
2022-08-01 yguyon Bump AppVeyor Visual Studio version to 2022
2022-07-28 wtc Shorten AppVeyor build to fix build timeout
2022-07-28 wtc Add int32_t casts to fix MSVC warnings
2022-07-25 ccom svt: null metadata field on EbBufferHeaderType
2022-07-15 vrabaud Adding a basic test for lossless. (#984)
2022-07-12 yguyon Use I420ToRGB24Matrix() in reformat_libyuv.c
2022-07-11 wtc Initialize base class avifRWData with {nullptr, 0}
2022-07-11 yguyon Rename YUVToRGB test suite to RGBToYUV

Created with:
  roll-dep src/third_party/libavif/src

Update two switch statements in avif_image_decoder.cc to handle the new
enum value 'AVIF_PIXEL_FORMAT_COUNT', otherwise there are -Wswitch
warnings.

R=dalecurtis@chromium.org,jzern@chromium.org

Change-Id: I303f40763bd6894be9be7f2c60807678fc4939f8
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3828790
Commit-Queue: Wan-Teh Chang <wtc@google.com>
Reviewed-by: James Zern <jzern@google.com>
Cr-Commit-Position: refs/heads/main@{#1035544}
---
 DEPS                                                            | 2 +-
 .../renderer/platform/image-decoders/avif/avif_image_decoder.cc | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/third_party/blink/renderer/platform/image-decoders/avif/avif_image_decoder.cc b/third_party/blink/renderer/platform/image-decoders/avif/avif_image_decoder.cc
index c24bf8e0446d3..6ccc4bcfe2c36 100644
--- a/third_party/blink/renderer/platform/image-decoders/avif/avif_image_decoder.cc
+++ b/third_party/blink/renderer/platform/image-decoders/avif/avif_image_decoder.cc
@@ -158,6 +158,7 @@ media::VideoPixelFormat AvifToVideoPixelFormat(avifPixelFormat fmt,
     case AVIF_PIXEL_FORMAT_YUV444:
       return kYUV444Formats[has_alpha][depth_index];
     case AVIF_PIXEL_FORMAT_NONE:
+    case AVIF_PIXEL_FORMAT_COUNT:
       NOTREACHED();
       return media::PIXEL_FORMAT_UNKNOWN;
   }
@@ -300,6 +301,7 @@ cc::YUVSubsampling AVIFImageDecoder::GetYUVSubsampling() const {
     case AVIF_PIXEL_FORMAT_YUV400:
       return cc::YUVSubsampling::kUnknown;
     case AVIF_PIXEL_FORMAT_NONE:
+    case AVIF_PIXEL_FORMAT_COUNT:
       NOTREACHED();
       return cc::YUVSubsampling::kUnknown;
   }
